src/encode.rs

```diff
@@ -1,463 +0,0 @@
-use crate::{
-    errors::{Error, Result},
-    key::PublicKeyParts,
-    parse::rsa_oid,
-    PublicKey, RSAPrivateKey, RSAPublicKey,
-};
-use num_bigint::{BigUint, ModInverse, ToBigInt};
-use num_traits::Zero;
-#[cfg(feature = "pem")]
-use pem::{EncodeConfig, LineEnding};
-use simple_asn1::{to_der, ASN1Block, BigInt};
-use std::prelude::v1::*;
-use std::{format, vec};
-
-const BYTE_BIT_SIZE: usize = 8;
-#[cfg(feature = "pem")]
-const DEFAULT_ENCODING_CONFIG: EncodeConfig = EncodeConfig {
-    line_ending: LineEnding::LF,
-};
-
-#[cfg(feature = "pem")]
-/// Trait for encoding the private key in the PEM format
-///
-/// Important: Encoding multi prime keys isn't supported. See [RustCrypto/RSA#66](https://github.com/RustCrypto/RSA/issues/66) for more info
-pub trait PrivateKeyPemEncoding: PrivateKeyEncoding {
-    const PKCS1_HEADER: &'static str;
-    const PKCS8_HEADER: &'static str = "PRIVATE KEY";
-
-    /// Converts a Private key into `PKCS1` encoded bytes in pem format.
-    ///
-    /// Encodes the key with the header:
-    /// `-----BEGIN <name> PRIVATE KEY-----`
-    ///
-    /// # Example
-    ///
-    /// ```
-    /// use rsa::{RSAPrivateKey, PrivateKeyPemEncoding};
-    /// use rand::rngs::OsRng;
-    ///
-    /// let mut rng = OsRng;
-    /// let bits = 2048;
-    /// let private_key = RSAPrivateKey::new(&mut rng, bits).expect("failed to generate a key");
-    ///
-    /// let _ = private_key.to_pem_pkcs1();
-    /// ```
-    fn to_pem_pkcs1(&self) -> Result<String> {
-        self.to_pem_pkcs1_with_config(DEFAULT_ENCODING_CONFIG)
-    }
-
-    /// Converts a Private key into `PKCS1` encoded bytes in pem format with encoding config.
-    ///
-    /// # Example
-    /// ```
-    /// # use rsa::{RSAPrivateKey, PrivateKeyPemEncoding};
-    /// # use rand::rngs::OsRng;
-    /// use pem::{EncodeConfig, LineEnding};
-    /// # let mut rng = OsRng;
-    /// # let bits = 2048;
-    /// # let private_key = RSAPrivateKey::new(&mut rng, bits).expect("failed to generate a key");
-    ///
-    /// let _ = private_key.to_pem_pkcs1_with_config(EncodeConfig {
-    ///     line_ending: LineEnding::CRLF,
-    /// });
-    /// ```
-    fn to_pem_pkcs1_with_config(&self, config: EncodeConfig) -> Result<String> {
```

Comment:
Is there a way to get the line encoding config with the pkcs1/pkcs8 crate?

Reply:
Currently a TODO, although not too hard to add implementation-wise: https://github.com/RustCrypto/utils/blob/master/pem-rfc7468/src/encoder.rs#L12-L14

Reply:
PR to add support for selecting line endings to `pem-rfc7468`:

https://github.com/RustCrypto/utils/pull/551

Reply:
Here's a PR to add line ending customization to `pkcs1`, using the newly released `pem-rfc7468`:

https://github.com/RustCrypto/utils/pull/553/files

If that looks good I can do `pkcs8` too.

Reply:
Went ahead and cut new releases of `pkcs1` and `pkcs8` with this feature added (0.2.3/0.7.5 respectively), and bumped them in a9de07d5313590d68353cdfb1b627f0488700377